src/_pytest/pytester.py

```diff
@@ -1492,13 +1495,17 @@ def fnmatch_lines(
         matches and non-matches are also shown as part of the error message.
 
         :param lines2: string patterns to match.
-        :param consecutive: match lines consecutive?
+        :param mode:
+            "consecutive": match lines consecutivly
+            "complete": require all lines to be matched in total
         """
         __tracebackhide__ = True
-        self._match_lines(lines2, fnmatch, "fnmatch", consecutive=consecutive)
+        self._match_lines(lines2, fnmatch, "fnmatch", mode=mode)
 
     def re_match_lines(
-        self, lines2: Sequence[str], *, consecutive: bool = False
+        self,
+        lines2: Sequence[str],
+        mode: Optional["Literal['consecutive', 'complete']"] = None,
```

Comment:
Perhaps instead of literals we should start to use `Enum` instead? It could be declared directly in LineMatcher for easy access:

```python
class LineMatcher:

    class MatchOptions(Enum):
        """Enum that contains all the various options that the *match* methods
        of this class use to handle the lines passed.
        
        :ivar Consecutive:
             Lines must be match consecutively 
        :ivar Complete:
             All input lines must be matched, with no unmatched lines left.
        """
        Consecutive = 0
        Complete = 1
```

This places a good place for docs and I suspect also helps with type hints.

Reply:
I think it overcomplicates things.  AFAICS `consecutive` will be released as a kwarg anyway soon, and there are not many other modes I can think of currently.